src/spec-configuration/containerFeaturesOrder.ts

```diff
@@ -3,133 +3,538 @@
+}
+
+async function applyOverrideFeatureInstallOrder(
+	params: CommonParams,
+	processFeature: (userFeature: DevContainerFeature) => Promise<FeatureSet | undefined>,
+	worklist: FNode[],
+	config: { overrideFeatureInstallOrder?: string[] },
+) {
+	const { output } = params;
+
+	if (!config.overrideFeatureInstallOrder) {
+		return worklist;
+	}
+
+	const overrideFeatureNodes: FNode[] = [];
+
+	// Create an artificial node for each Feature in the override property.
+	// Reverse iterate to remove out items from config that have been processed.
+	for (let j = config.overrideFeatureInstallOrder.length - 1; j >= 0; j--) {
+		const userFeatureId = config.overrideFeatureInstallOrder[j];
+		const overrideNode: FNode = {
+			type: 'artificial',
+			userFeatureId,
+			options: {},
+			dependsOn: [],
+			installsAfter: [],
+		};
+		const processed = await processFeature(overrideNode);
+		if (!processed) {
+			throw new Error(`Feature '${userFeatureId}' in 'overrideFeatureInstallOrder' could not be processed.`);
+		}
+
+		if (featureSupportsDependencies(processed)) {
+			overrideNode.featureSet = processed;
+			overrideFeatureNodes.push(overrideNode);
+			config.overrideFeatureInstallOrder.splice(j, 1);
+		}
+	}
+
+	overrideFeatureNodes.reverse();
+	output.write(`[override-feature-nodes]: ${overrideFeatureNodes.map(n => n.userFeatureId).join(', ')}`, LogLevel.Info);
+
+	for (let i = 0; i < overrideFeatureNodes.length - 1; i++) {
+		// Scan the resolvedGraph for a node that matches the next override node.
+		const currentOverrideNode = overrideFeatureNodes[i];
+		const nextOverrideNode = overrideFeatureNodes[i + 1];
+		let countMatches = 0;
+		for (const node of worklist) {
+			if (satisfiesSoftDependency(node, nextOverrideNode)) {
+				countMatches++;
+				// Add a soft dependency on the next override node.
+				output.write(`[override-feature-add-relationship]: ${node.userFeatureId} installsAfter ${currentOverrideNode.userFeatureId}`, LogLevel.Info);
+				node.installsAfter.push(currentOverrideNode);
+			}
+		}
+
+		// Invariant: Every Feature indicated in the 'overrideFeatureInstallOrder' property should have at least one corresponding Feature in the resolved graph.
+		if (!countMatches) {
+			throw new Error(`The 'overrideFeatureInstallOrder' property includes references to Feature(s) that are not queued for installation.`);
+		}
+	}
+
+	// Add the final node to each leaf node's soft-dependency adjacency list (Enforcing override Features are installed first),
+	// Ensuring that such an act does not introduce a cycle.
+	const lastOverrideNode = overrideFeatureNodes[overrideFeatureNodes.length - 1];
+	for (const node of worklist) {
+		if (node.installsAfter.length === 0 //  TODO: necessary?
+			&& node.dependsOn.length === 0 // Leaf node
+			&& !satisfiesSoftDependency(node, lastOverrideNode) // Prevent cycles with itself
+			&& overrideFeatureNodes.every(override => !satisfiesSoftDependency(node, override))) // Prevent cycles with other override nodes 
```

Comment:
This only detects cycles with 2 nodes involved. We detect cycles later and report them. Should we leave the handling to the user? (A few lines up we don't try to detect them either, but these additons to `installsAfter` could also introduce additional cycles.)

Reply:
You suggest removing this check entirely, as we won't be able to detect all cycles at this stage anyway?

Reply:
This code was removed.